docs/getting-started/src/getting-started/counter/deploy-and-solve.md

```diff
@@ -0,0 +1,172 @@
+# Deploy & Solve
+
+Now that we've built our simple counter contract, let's deploy it to a local
+test network, and update the counter by solving our `Increment` predicate.
+
+Before we begin, we'll make the `essential-builder` and `essential-rest-client`
+tools available to our current shell:
+
+```
+nix shell github:essential-contributions/essential-integration#essential
+```
+
+## Running a Test Builder
+
+Before we can deploy our contract, we need somewhere to deploy it to. Let's run
+a local, in-memory test instance of the `essential-builder` tool:
+
+```console
+essential-builder --validation
+```
+
+Here we're running a local instance of the builder with node validation
+enabled. We should see some output like the following:
+
+```console
+2024-11-14T11:58:14.359260Z  INFO essential_builder_cli: Initializing node DB
+2024-11-14T11:58:14.364424Z  INFO essential_builder_cli: Starting node API server at 0.0.0.0:57967
+2024-11-14T11:58:14.364439Z  INFO essential_builder_cli: Initializing builder DB
+2024-11-14T11:58:14.365863Z  INFO essential_builder_cli: Starting builder API server at 0.0.0.0:57968
+2024-11-14T11:58:14.365887Z  INFO essential_builder_cli: Running the block builder
+```
+
+We can see that the builder exposes 2 APIs:
+
+1. **A node API** at **localhost** on **port 57967**. This allows for querying
+   the state of our local single-node blockchain.
+2. **A builder API** at **localhost** on **port 57968**. This allows for
+   submitting solutions to have them included in a block.
+
+
+## Contract Deployment
+
+Using the `essential-rest-client` tool, we can deploy our built counter
```

Comment:
Should we mentioned that this should be done in separate terminal and that for that terminal, we also need the `nix` command from earlier?